src/javascript/binary/common_functions/sparkline_chart.js

```diff
@@ -0,0 +1,130 @@
+const getHighstock = require('./common_functions').requireHighstock;
+
+let Highcharts;
+const sparkline_options = {
+    chart: {
+        type  : 'spline',
+        margin: [2,0,2,0],
+        height: 30,
+        style : { overflow: 'visible' },
+        boost : {
+            seriesThreshold   : 1, // Enable by default
+            useGPUTranslations: true, // For better boost
+        },
+    },
+    title    : { text: '' },
+    credits  : { enabled: false },
+    exporting: { enabled: false },
+    legend   : { enabled: false },
+    xAxis    : {
+        labels            : { enabled: false },
+        title             : { text: null },
+        startOnTick       : false,
+        lineWidth         : 0,
+        minorGridLineWidth: 0,
+        lineColor         : 'transparent',
+        gridLineColor     : 'transparent',
+        endOnTick         : false,
+        tickPositions     : [],
+    },
+    yAxis: {
+        endOnTick         : false,
+        startOnTick       : false,
+        labels            : { enabled: false },
+        lineWidth         : 0,
+        minorGridLineWidth: 0,
+        lineColor         : 'transparent',
+        gridLineColor     : 'transparent',
+        title             : { text: null },
+        tickPositions     : [0],
+    },
+    tooltip: {
+        formatter() {
+            return `<span>${this.y}</span>`;
+        },
+        backgroundColor: null,
+        borderWidth    : 0,
+        shadow         : false,
+        useHTML        : true,
+        hideDelay      : 1,
+        shared         : true,
+        padding        : 0,
+        positioner(w, h, point) {
+            return { x: point.plotX - (w / 2), y: point.plotY - h };
+        },
+    },
+    plotOptions: {
+        series: {
+            animation: false,
+            lineWidth: 1,
+            shadow   : false,
+            states   : {
+                hover: { lineWidth: 1 },
+            },
+            marker: {
+                radius : 2,
+                enabled: false,
+            },
+            fillOpacity: 0.25,
+        },
+    },
+    series: [{
+        name: 'Price',
+        data: [],
+    }],
+};
+
+// Initialize Highcharts.
+getHighstock((H) => {
```

Comment:
This causes to load the highstock js file on all pages. Needs to be called on demand.
Maybe moving it inside the `showSparkline` and use as a promise?